src/Raven.Server/Documents/TimeSeries/TimeSeriesValuesSegment.cs

```diff
@@ -126,110 +126,119 @@ public static void AssertValueStatus(ulong status)
-                    // need to write the tag
-                    insertTag = true;
+                    else
+                    {
+                        // need to write the tag
+                        insertTag = true;
 
-                    tempBitsBuffer.AddValue(3, 2); // will write the tag index buffer here
-                    tempBitsBuffer.AddValue((ulong)(~prevTagIndex), 7);
-                    tempHeader->PreviousTagIndex = (byte)(~prevTagIndex);
-                }
+                        tempBitsBuffer.AddValue(3, 2); // will write the tag index buffer here
+                        tempBitsBuffer.AddValue((ulong)(~prevTagIndex), 7);
+                        tempHeader->PreviousTagIndex = (byte)(~prevTagIndex);
+                    }
 
-                tempHeader->PreviousTimestamp = deltaFromStart;
-                tempHeader->NumberOfEntries++;
-                var actualBitsBuffer = GetBitsBuffer(tempHeader);
+                    tempHeader->PreviousTimestamp = deltaFromStart;
+                    tempHeader->NumberOfEntries++;
+                    var actualBitsBuffer = GetBitsBuffer(tempHeader);
 
-                if (insertTag)
-                {
-                    if (actualBitsBuffer.EnsureAdditionalBits(allocator, tempBitsBuffer.NumberOfBits + (tag.Length + 1) * 8) == false)
-                        return false;
+                    if (insertTag)
+                    {
+                        if (actualBitsBuffer.EnsureAdditionalBits(allocator, tempBitsBuffer.NumberOfBits + (tag.Length + 1) * 8) == false)
+                            return false;
 
-                    InsertTag(tag, tempHeader, actualBitsBuffer.NumberOfBytes);
-                    actualBitsBuffer = GetBitsBuffer(tempHeader);
-                    var result = actualBitsBuffer.AddBits(allocator, tempBitsBuffer);
-                    Debug.Assert(result, "Failed to add bits, but already checked precondition");
-                }
-                else
-                {
-                    if (actualBitsBuffer.AddBits(allocator, tempBitsBuffer) == false)
+                        InsertTag(tag, tempHeader, actualBitsBuffer.NumberOfBytes);
+                        actualBitsBuffer = GetBitsBuffer(tempHeader);
+                        var result = actualBitsBuffer.AddBits(allocator, tempBitsBuffer);
+                        Debug.Assert(result, "Failed to add bits, but already checked precondition");
+                    }
+                    else
                     {
-                        return false;
+                        if (actualBitsBuffer.AddBits(allocator, tempBitsBuffer) == false)
+                        {
+                            return false;
+                        }
                     }
-                }
 
-                Memory.Copy(Header, tempHeader, copiedHeaderSize);
+                    Memory.Copy(Header, tempHeader, copiedHeaderSize);
 
-                return true;
+                    return true;
```

Comment:
Look at lines 216 here, we are doing explicit checks on the temp buffer, so this shouldn't be the case.
How can you get this error?

Reply:
This check is too late... either the assert kills me or an `ArgumentException`/`IndexOutOfRangeException` when I'm trying to add to the `tempBitBuffer` where there is not space left.